pandas/core/indexing.py

```diff
@@ -1737,22 +1744,26 @@ def _getitem_scalar(self, key):
         values = self.obj._get_value(*key)
         return values
 
-    def _get_partial_string_timestamp_match_key(self, key, labels):
+    def _get_partial_string_timestamp_match_key(self, key, labels: Index):
         """Translate any partial string timestamp matches in key, returning the
         new key (GH 10331)"""
         if isinstance(labels, MultiIndex):
             if isinstance(key, str) and labels.levels[0].is_all_dates:
                 # Convert key '2016-01-01' to
                 # ('2016-01-01'[, slice(None, None, None)]+)
-                key = tuple([key] + [slice(None)] * (len(labels.levels) - 1))
+                list_items = [key]  # type: List[Union[slice,str]]
+                list_items += [slice(None)] * (len(labels.levels) - 1)
+                key = tuple(list_items)
```

Comment:
no happy changing code. the original is perfectly valid. will probably revert this and add a type: ignore and wait for a mypy update.

@WillAyd should we add `--warn-unused-ignores` to either the CI, or `warn-unused-ignores=True` to the ini file.

or happy to just run ad-hoc periodically.

Reply:
Can we not reuse `key` and just call it something else?

Reply:
the issue is that you can't add a list of slices to a list of strings in an inline expression.

Reply:
Ah ok that's strange. Let me open an issue on Mypy tracker and we can ref that in ignore comment here

Reply:
Actually I think the slice is a red herring. This fails with just int and str as well:

```python
values: Union[List[int], List[str]] = [1]  + [""]
# error: List item 0 has incompatible type "str"; expected "int"

values: List[Union[int, str]] = [1]  + [""]
# error: Incompatible types in assignment (expression has type "List[int]", variable has type "List[Union[int, str]]")
# note: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
# note: Consider using "Sequence" instead, which is covariant
# error: List item 0 has incompatible type "str"; expected "int"
```

So I think

Reply:
see https://github.com/python/mypy/issues/5492

Reply:
Yea good find. Can just add that as comment and ignore (suggested approach from mypy for dealing with these types of things)